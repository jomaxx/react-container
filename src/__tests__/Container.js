import React from "react";
import { render } from "react-testing-library";
import { Container } from "../";

it("renders container", () => {
  const spy = jest.fn(() => null);

  render(<Container>{spy}</Container>);

  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toMatchSnapshot();
});

it("consumes container", () => {
  const spy = jest.fn(() => null);

  render(
    <Container>
      <Container.Consumer>{spy}</Container.Consumer>
    </Container>
  );

  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toMatchSnapshot();
});

it("sets state", () => {
  const spy = jest.fn(() => null);

  render(
    <Container>
      <Container.Consumer>{spy}</Container.Consumer>
    </Container>
  );

  spy.mock.calls[0][0].setState({ done: true });

  expect(spy).toHaveBeenCalledTimes(2);
  expect(spy).toMatchSnapshot();
});
