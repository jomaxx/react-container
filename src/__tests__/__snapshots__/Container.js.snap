// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`consumes container 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "setState": [Function],
        "state": Object {},
      },
    ],
  ],
}
`;

exports[`renders container 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "setState": [Function],
        "state": Object {},
      },
    ],
  ],
}
`;

exports[`sets state 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "setState": [Function],
        "state": Object {},
      },
    ],
    Array [
      Object {
        "setState": [Function],
        "state": Object {
          "done": true,
        },
      },
    ],
  ],
}
`;
